arch/Kconfig

```diff
@@ -715,6 +715,20 @@ config FPU_SHARING
 	  across context switches to allow multiple threads to perform concurrent
 	  floating point operations.
 
+config FP_USE_SOFT_FLOAT
```

Comment:
Could we try to get soft float configuration aligned ?

In arm, there is already:
https://github.com/zephyrproject-rtos/zephyr/blob/master/arch/arm/core/aarch32/Kconfig#L217-L235

also, should it not be possible to select soft float even if you have a fpu ?
Ther could be reasons for still compile with softfloat, e.g. for testing.

Reply:
Removed `depends on !FPU`.

AFAIK, the current `CONFIG_FP_SOFTABI` in ARM is not exactly the same as `-msoft-float`. Using `-msoft-float` tells GCC to emit code to use the soft float library directly and would not emit any FPU instructions. `CONFIG_FP_SOFTABI` (and thus `-mfloat-abi=softfp`) dictates the calling conventions on floating point arguments, where they are passed via integer registers instead of FP registers. GCC is allowed to emit FPU instructions if so desired. So, strictly speaking, `CONFIG_FP_SOFTABI` may not do floating points entirely in software. I am not sure how to consolidate all these together. Suggestions are welcomed.

Reply:
> the current CONFIG_FP_SOFTABI in ARM is not exactly the same as -msoft-float. Using -msoft-float tells GCC to emit code to use the soft float library directly and would not emit any FPU instructions. CONFIG_FP_SOFTABI (and thus -mfloat-abi=softfp) dictates the calling conventions on floating point arguments, where they are passed via integer registers instead of FP registers. 

Right.
The error messages from gcc actually indicates that `-msoft-float` == `-mfloat-abi=soft`, cause if you try to compile a source file using: `-msoft-float -mfloat-abi=hard`, then gcc will return the following error.
```
error: -mfloat-abi=soft and -mfloat-abi=hard may not be used together
```
even though `-mfloat-abi=soft` was never specified as argument.
That can either be because gcc actually use `-mfloat-abi=soft` when you specify `-msoft-float` or because `-mfloat-abi=soft` is implied with `-msoft-float`.

From reading the docs I haven't been able to determine which of the above is actually the case.

Either way, it seems `-msoft-float` should be restricted with `FP_HARDABI=y`, and might be redundant when specifying `-mfloat-abi=soft`.
Gcc docs seems vague on this part.

Reply:
Added `depends on` for the kconfig, where this option is only visible with `FP_HARDABI=n`.